x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorStateMachine.java

```diff
@@ -56,6 +58,9 @@ public static void assertValidStateTransition(ConnectorStatus current, Connector
     }
 
     public static Set<ConnectorStatus> validNextStates(ConnectorStatus current) {
-        return VALID_TRANSITIONS.getOrDefault(current, Collections.emptySet());
+        Set<ConnectorStatus> nextStates = new HashSet<>(VALID_TRANSITIONS.getOrDefault(current, Collections.emptySet()));
+        // Transition to the same state is allowed
+        nextStates.add(current);
```

Comment:
Why not just add the explicit states in VALID_TRANSITIONS?
This obfuscates what are actually valid transitions by splitting the logic between that literal map and this function.